snforge_std/src/cheatcodes.cairo

```diff
@@ -39,6 +39,42 @@ impl RevertedTransactionImpl of RevertedTransactionTrait {
     }
 }
 
+#[derive(Drop)]
+struct TxInfoMock {
+    version: Option<felt252>,
+    account_contract_address: Option<felt252>,
+    max_fee: Option<u128>,
+    signature: Option<Array<felt252>>,
+    transaction_hash: Option<felt252>,
+    chain_id: Option<felt252>,
+    nonce: Option<felt252>,
+}
+
+trait TxInfoMockTrait {
+    fn default() -> TxInfoMock;
+}
+
+impl TxInfoMockImpl of TxInfoMockTrait {
+    fn default() -> TxInfoMock {
+        TxInfoMock {
+            version: Option::None(()),
+            account_contract_address: Option::None(()),
+            max_fee: Option::None(()),
+            signature: Option::None(()),
+            transaction_hash: Option::None(()),
+            chain_id: Option::None(()),
+            nonce: Option::None(()),
+        }
+    }
+}
+
+fn is_set_and_value<T, impl TDrop: Drop::<T>>(option: Option<T>, default: T) -> (bool, T) {
```

Comment:
The naming is somewhat confusing. Also, please consider moving it closer to usage